pkg/sif/create.go

```diff
@@ -216,55 +218,106 @@ func writeHeader(fimg *FileImage) error {
 	return nil
 }
 
-// CreateContainer is responsible for the creation of a new SIF container
-// file. It takes the creation information specification as input
-// and produces an output file as specified in the input data.
-func CreateContainer(cinfo CreateInfo) (fimg *FileImage, err error) {
-	fimg = &FileImage{}
-	fimg.DescrArr = make([]Descriptor, DescrNumEntries)
+// createOpts accumulates container creation options.
+type createOpts struct {
+	ID         uuid.UUID
+	InputDescr []DescriptorInput
+	GetTime    func() time.Time
+}
+
+// CreateOpt are used to specify container creation options.
+type CreateOpt func(*createOpts) error
+
+// WithID specifies id as the unique ID.
+func WithID(id string) CreateOpt {
+	return func(co *createOpts) error {
+		id, err := uuid.Parse(id)
+		co.ID = id
+		return err
+	}
+}
+
+// WithDescriptors appends dis to the list of descriptors.
+func WithDescriptors(dis ...DescriptorInput) CreateOpt {
+	return func(co *createOpts) error {
+		co.InputDescr = append(co.InputDescr, dis...)
+		return nil
+	}
+}
+
+// WithTimeFunc specifies fn as the function to obtain timestamps.
+func WithTimeFunc(fn func() time.Time) CreateOpt {
+	return func(co *createOpts) error {
+		co.GetTime = fn
+		return nil
+	}
+}
+
+// CreateContainer creates a new SIF container file at path, according to opts.
+func CreateContainer(path string, opts ...CreateOpt) (*FileImage, error) {
+	id, err := uuid.NewRandom()
+	if err != nil {
+		return nil, err
+	}
+
+	co := createOpts{
+		ID:      id,
+		GetTime: time.Now,
+	}
+
+	for _, opt := range opts {
+		if err := opt(&co); err != nil {
+			return nil, err
+		}
+	}
+
+	now := co.GetTime()
```

Comment:
Tiny nit... use of the word `now` here, when the `now` function might well not return a time that is not now. Any reason not to just use `GetTime().Unix()` more directly below?

Reply:
> Any reason not to just use GetTime().Unix() more directly below?

Trying to make it easy to answer the question "has the image been modified since it was created"... which isn't easy if two calls to `GetTime()` are made.

I'll switch out `now` for `t`... that's more generic and accurate if/when `GetTime` is returning something other than the current time.